CRM/Core/BAO/CustomField.php

```diff
@@ -392,6 +393,10 @@ public static function &getFields(
         if ($onlyParent) {
           $extends .= " AND $cgTable.extends_entity_column_value IS NULL AND $cgTable.extends_entity_column_id IS NULL ";
         }
+        // Temporary hack - in 5.27 a new field is added to civicrm_custom_field. There is a high
+        // risk this function is called before the upgrade page can be reached and if
+        // so it will potentially result in fatal error.
+        $serializeField = CRM_Core_BAO_Domain::isDBUpdateRequired() ? '' : "$cfTable.serialize,";
```

Comment:
what about using instead @eileenmcnaughton 

```suggestion
        $serializeField = CRM_Core_DAO::checkFieldExists('civicrm_custom_field', 'serialize') ? '' : "$cfTable.serialize,";
```

Reply:
I think the one I used would be cached actually

Reply:
(in general schema queries are more likely to cause locks too I think)

Reply:
My understanding was that the checkFieldExists is what we have been using in these sorts of things

Reply:
@seamuslee001 hmm - mostly in upgrade scripts not in functions that could be hit often, even in upgraded dbs

Reply:
@eileenmcnaughton you can see its included in other places e.g. in the ConfigSetting.php and Menu.php https://github.com/civicrm/civicrm-core/search?p=2&q=checkIfFieldExists&unscoped_q=checkIfFieldExists so its not just about upgrade scripts

Reply:
@seamuslee001 it probably shouldn't be used in those places!